lib/Parse/ParsePattern.cpp

```diff
@@ -373,8 +377,16 @@ Parser::parseParameterClause(SourceLoc &leftParenLoc,
             // SE-110 depending on the kind of declaration.  We now need to
             // warn about the misuse of this syntax and offer to
             // fix it.
-            diagnose(typeStartLoc, diag::parameter_unnamed_warn)
-              .fixItInsert(typeStartLoc, "_: ");
+            // An exception to this rule is when the type is declared with type sugar
+            // Reference: SR-11724
+            if ( isa<OptionalTypeRepr>(param.Type)
```

Comment:
Nit: A little bit of extra whitespace seems to have slipped in.

Reply:
Done